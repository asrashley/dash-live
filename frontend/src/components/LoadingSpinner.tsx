
export function LoadingSpinner() {
    return <div className="lds-ring">
    <div className="lds-seg" />
    <div className="lds-seg" />
    <div className="lds-seg" />
    <div className="lds-seg" />
    </div>;
}

