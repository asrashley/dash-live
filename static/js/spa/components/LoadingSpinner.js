import { html } from 'htm/preact';

export function LoadingSpinner() {
    return html`<div className="lds-ring">
    <div className="lds-seg" />
    <div className="lds-seg" />
    <div className="lds-seg" />
    <div className="lds-seg" />
    </div>`;
}

