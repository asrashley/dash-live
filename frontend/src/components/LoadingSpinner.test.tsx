import { describe, expect, test } from "vitest";

import { renderWithProviders } from "../test/renderWithProviders";
import { LoadingSpinner } from "./LoadingSpinner";

describe("LoadingSpinner", () => {
  test("should display spinner", () => {
    const { asFragment, getBySelector, getAllBySelector } = renderWithProviders(
      <LoadingSpinner  />
    );
    getBySelector('.lds-ring');
    expect(getAllBySelector('.lds-ring > .lds-seg').length).toEqual(4);
    expect(asFragment()).toMatchSnapshot();
  });
});
