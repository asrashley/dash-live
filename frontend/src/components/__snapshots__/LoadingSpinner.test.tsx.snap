// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner > should display spinner 1`] = `
<DocumentFragment>
  <div
    class="lds-ring"
  >
    <div
      class="lds-seg"
    />
    <div
      class="lds-seg"
    />
    <div
      class="lds-seg"
    />
    <div
      class="lds-seg"
    />
  </div>
</DocumentFragment>
`;
